puffin_egui/src/flamegraph.rs

```diff
@@ -712,38 +748,39 @@ fn to_ms(ns: NanoSecond) -> f64 {
 }
 
 fn paint_scope(
-    info: &Info,
+    info: &Info<'_>,
     options: &mut Options,
     stream: &Stream,
     scope: &Scope<'_>,
     depth: usize,
     min_y: f32,
-) -> Result<PaintResult> {
+) -> Result<Option<PaintResult>> {
     let top_y = min_y + (depth as f32) * (options.rect_height + options.spacing);
 
-    let result = paint_record(info, options, "", "", &scope.record, top_y);
+    let result = paint_record(info, options, "", "", scope.id, &scope.record, top_y);
 
-    if result != PaintResult::Culled {
+    if !matches!(result, Some(PaintResult::Culled)) {
         let mut num_children = 0;
         for child_scope in Reader::with_offset(stream, scope.child_begin_position)? {
-            paint_scope(info, options, stream, &child_scope?, depth + 1, min_y)?;
+            let _ = paint_scope(info, options, stream, &child_scope?, depth + 1, min_y)?;
```

Comment:
im not ignoring the result but the option. We return Ok(None) when the id does not have scope details, returning option because the error is about serialisation stuff and wasn't sure if its appropriated to add a new error there for when missing scope details for scope id.

Reply:
I always use clippy lint that enforced declaring a return type even if it isn't used. After this PR I'll add more lints to catch those cases.